src/modules/kvs/commit.c

```diff
@@ -209,14 +209,17 @@ static int store_cache (commit_t *c, int current_epoch, json_t *o,
     return rc;
 }
 
-static bool fileval_big (json_t *value)
+/* -1 on error, 0 on false, 1 on true */
+static int fileval_big (json_t *value)
 {
     char *s = json_dumps (value, JSON_ENCODE_ANY);
-    if (!s)
-        oom ();
+    if (!s) {
+        errno = ENOMEM;
+        return -1;
+    }
     bool rc = strlen (s) > BLOBREF_MAX_STRING_SIZE;
     free (s);
-    return rc;
+    return rc ? 1 : 0;
```

Comment:
Ahhh, I see what you mean.  Yeah, that'll probably clarify things a bit.  But more importantly it'll use ```kvs_util_json_dumps()```, which I had missed earlier.